src/plugins/dashboard/public/url_generator.ts

```diff
@@ -135,15 +142,21 @@ export const createDashboardUrlGenerator = (
       `${appBasePath}#/${hash}`
     );
 
-    return setStateToKbnUrl<QueryState>(
+    url = setStateToKbnUrl<QueryState>(
       GLOBAL_STATE_STORAGE_KEY,
       cleanEmptyKeys({
         time: state.timeRange,
         filters: filters?.filter((f) => esFilters.isFilterPinned(f)),
         refreshInterval: state.refreshInterval,
       }),
       { useHash },
-      appStateUrl
+      url
     );
+
+    if (state.searchSessionId) {
+      url = `${url}&${DashboardConstants.SEARCH_SESSION_ID}=${state.searchSessionId}`;
```

Comment:
//nit 
We don't have any function in our code base to serialize url params?

Reply:
I think that would be 
`import { stringify } from 'query-string';` but I guess in this particular case using it would add more code then not using it